hydroflow/tests/surface_codegen.rs

```diff
@@ -114,6 +115,77 @@ pub fn test_large_diamond() {
     df.run_available();
 }
 
+#[test]
+pub fn test_split() {
+    use hydroflow::{tl, tt};
+
+    let (out_send_a, mut out_recv_a) = hydroflow::util::unbounded_channel::<usize>();
+    let (out_send_b, mut out_recv_b) = hydroflow::util::unbounded_channel::<usize>();
+
+    fn split_even_odd(x: usize) -> tt!(Option<usize>, Option<usize>) {
+        if 0 == x % 2 {
+            tl!(Some(x), None)
+        } else {
+            tl!(None, Some(x))
+        }
+    }
+
+    let mut df = hydroflow_syntax! {
+        s = recv_iter(0..10) -> map(split_even_odd) -> split();
+        s[0] -> flatten() -> for_each(|x| out_send_a.send(x).unwrap());
+        s[1] -> flatten() -> for_each(|x| out_send_b.send(x).unwrap());
+    };
+    df.run_available();
+
+    let out_a: Vec<_> = recv_into(&mut out_recv_a);
+    let out_b: Vec<_> = recv_into(&mut out_recv_b);
+    assert_eq!(&[0, 2, 4, 6, 8], &*out_a);
+    assert_eq!(&[1, 3, 5, 7, 9], &*out_b);
+}
+
+#[test]
+pub fn test_split_enum() {
+    #[derive(Split, Debug)]
+    pub enum Value {
+        Integer(u32),
+        String(&'static str),
+        Null,
+        Point { x: f32, y: f32 },
+    }
+
+    let (in_send, in_recv) = hydroflow::util::unbounded_channel::<Value>();
+
+    let (out_send_integer, mut out_recv_integer) = hydroflow::util::unbounded_channel::<u32>();
```

Comment:
This example seems unnecessarily complicated and even could be seen as an anti-pattern -- just because we have an enum-based split doesn't mean we need a channel for each branch. I realized on closer read that the channels are entirely a red herring. But I'd be happier if the reference code here didn't use extra channels, just split the inbound channel and did something with each branch (like add 1 to the integer, copy the string, etc. -- and then maybe print?)